deps/verifier/src/cca/remote.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023 Arm Ltd.
+// Copyright (c) 2025 Linaro Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::*;
+use base64::Engine;
+use config::Config;
+use core::result::Result::Ok;
+use ear::Ear;
+use log::{debug, error};
+use std::str;
+
+const MEDIA_TYPE: &str = r#"application/eat-collection; profile="http://arm.com/CCA-SSD/1.0.0""#;
+
+fn evidence_builder(
+    nonce: &[u8],
+    accept: &[String],
+    token: Vec<u8>,
+) -> Result<(Vec<u8>, String), veraison_apiclient::Error> {
+    debug!("server challenge: {:?}", nonce);
+    debug!("acceptable media types: {:#?}", accept);
+    // TODO: Get the CCA media type from the slice of `accept`.
+    Ok((token, MEDIA_TYPE.to_string()))
+}
+
+pub async fn verify(
+    config: Config,
+    token: &[u8],
+    expected_report_data: &Vec<u8>,
+) -> Result<Ear, anyhow::Error> {
+    let config::CcaVerifierConfig::Remote { address, ca_cert } = config.cca_verifier else {
+        panic!("expecting remote configuration")
+    };
+
+    let mut disco_builder = DiscoveryBuilder::new().with_base_url(address.clone());
+
+    if ca_cert.is_some() {
+        disco_builder = disco_builder.with_root_certificate(ca_cert.clone().unwrap())
+    }
+
+    let disco = disco_builder
+        .build()
+        .context("Failed to start API discovery with the service")?;
+
+    let verification_api = disco
+        .get_verification_api()
+        .await
+        .context("Failed to discover the verification endpoint details")?;
+
+    let relpath = verification_api
+        .get_api_endpoint("newChallengeResponseSession")
+        .context("Could not locate a newChallengeResponseSession endpoint")?;
+
+    let api_endpoint = format!("{}{}", address, relpath);
+
+    let mut cr_builder = ChallengeResponseBuilder::new().with_new_session_url(api_endpoint);
+
+    if ca_cert.is_some() {
+        cr_builder = cr_builder.with_root_certificate(ca_cert.clone().unwrap())
+    }
+
+    let cr = cr_builder
+        .build()
+        .context("Failed to start challenge-response API with the service")?;
+
+    let n = Nonce::Value(expected_report_data.clone());
```

Comment:
Let me double-check what's going on here. Nonce could mean a couple different things. It could be a value that is used to make sure that the token we get from veraison is fresh. If this were the case I would expect it to be picked randomly here and not related to the `expected_report_data` although maybe you could reuse that value.

The `expected_report_data` is to be compared against the report data in the hardware evidence (idk the name for this with ARM) to make sure that the key + nonce use for the rcar handshake are bound to the hw evidence and can't be changed out.

It's a little unclear which of these two things is happening.

Reply:
TL; DR.  It’s the nonce in the token submitted for appraisal.

A slightly longer story: Veraison appraisal API has two “modes”: 

* Veraison is the challenger:
```http
POST /challenge-response/v1/newSession?nonceSize=64
```

* The relying party (in this case, Trustee) is the challenger:
```http
POST /challenge-response/v1/newSession?nonce=base64_encoded_value
```

In this case, we use the latter since the RP has already chosen the challenge (`expected_report_data`) and needs to tell Veraison which value to check in the submitted evidence.